TechnicalDocuments/Proposals/BabylonDesignLibrary.md

```diff
@@ -0,0 +1,120 @@
+# Babylon Design Library
+
+* Author(s): Giorgos Tsiapaliokas
+* Review Manager: ReviewManager
+
+## Introduction
+This proposal is about introducing a Design Library.
+
+## Motivation
+With our current infrastrure we have have reusable components which don't reuse the same style,
+we provide different styesheets based on the design.
+
+The idea behind this is to have reusable components with a predefined stylesheet displaying different data.
+The benefit of having a design library is
+1. to develop new UIs faster
+2. to catch inconsistencies in the UI due to a deisgn flaw or due to an engineering mistake.
+
+The requirement is to cover most of the common use cases and *not* to extend to edge cases.
+Also we must reuse our current tools like `Bento`
+
+## Proposed solution
+We can introduce a new service which will be responsible for creating these components following
+the semantics of the design styleguide which will be provided to us.
+
+The purpose of this service isn't to expose the full potential of `Bento` components but to follow the
+given design guidelines. This is an example.
+Also with this service we want to limit the exposure of specific components, in order to be able to introduce  
+drastic design changes in the future as easy as possible.
+
+Furthermore, this service can also host alerts, custom viewcontroller animations, etc. 
+
+``` swift
+final class AppearenceService {
```

Comment:
`BabylonAppAppearance` has been setup to serve a similar role `AppearenceService.Appearence`, and all `BabylonBoxViewController`s are set to observe `BabylonAppAppearance.shared`. Is there any reason why we should diverge from it?

Reply:
I didn't mention anything regarding the integration of `AppearenceService` to our current infrastructure because in my opinion this should be a different discussion.

The current proposal has been made in order to evaluate the need of having something like `AppearenceService`.
I would like to have this conversation in another proposal or PR.

Still this is a very good question regarding why `AppearenceService` is implemented in this proposal under its current form. 
The purpose of `AppearenceService` is to provide `AppearenceService.Appearence` without using a singleton or something like `NotificationCenter`.
Since `Current` has been introduced to our codebase injecting dependencies has become easier.

Also neither `AppearenceService` nor `AppearenceService.Appearence` depend on system behaviours (like `UITraitCollection`) because their purpose is to offer a brand mechanism which won't be restricted just to `Bento` components.

Reply:
This proposal does not touch on how appearance will be accessed by the renderers. I think it is required to move forward.

Reply:
I'm also concerned that this does not touch on how this will go along with existing `BabylonAppAppearance`, injecting yet another thing in all Bento renderers does not seem like a good idea to me.

I think `BabylonAppAppearance` could be explored in a direction of adapting it. I'm particularly not a fan of `ModernPallete`, so would like to understand how it can be replaced with something (it might be just a name change + integration with colours/fonts which are currently hardcoded into it)

Reply:
From my view, this is more or less a replica of the appearance system that has been put in place. If you drop everything inside `BabylonAppAppearance` and put in everything introduced here, we end up with the same outcome. 

Moreover, as @ilyapuchka said, ultimately it is about how things propagate into actual screens, and we should not treat it as a separate, independent work.

On an ending note, whatever _inside_ BabylonAppAppearance can be sidelined and deprecated, and how it is populated *now* does not imply it is desirable & extensions should follow it. It is merely born as a mean to share HC and CM styles in the first place.